src/components/DataTable.vue

```diff
@@ -0,0 +1,208 @@
+        v-for="col of columns" 
+        :key="col.field" 
+        :header="col.header" 
+        :field="col.field"
+        :dataType="col.dataType"
+        :sortable="(col.sort !== false)"
+        :showFilterMatchModes="!col.useMultiSelect"
+      >
+        <template v-if="col.dataType === 'date'" #body="{ data }">
+          {{ getFormattedDate(data[col.field]) }}
+        </template>
+        <template v-if="col.dataType" #filter="{ filterModel }">
+          <InputText 
+            v-if="col.dataType === 'text' && !col.useMultiSelect"
+            type="text" v-model="filterModel.value" 
+            class="p-column-filter" 
+            placeholder="Search" 
+          />
+          <MultiSelect 
+            v-if="col.useMultiSelect"
+            v-model="filterModel.value" 
+            :options="refOptions[col.field]" 
+            placeholder="Any"
+            :showToggleAll="false" 
+            class="p-column-filter" 
+          />
+          <Calendar 
+            v-if="col.dataType === 'date' && !col.useMultiSelect"
+            v-model="filterModel.value" 
+            dateFormat="mm/dd/yy" 
+            placeholder="mm/dd/yyyy" 
+          />
+        </template>
+      </Column>
+    </DataTable>
+  </div>
+</template>
+<script setup>
+import { ref } from 'vue';
+import { FilterMatchMode, FilterOperator } from "primevue/api";
+import SkeletonTable from "@/components/SkeletonTable.vue"
+import _get from 'lodash/get'
+import _map from 'lodash/map'
+import _forEach from 'lodash/forEach'
+import _find from 'lodash/find'
+import _filter from 'lodash/filter'
+import _toUpper from 'lodash/toUpper'
+import _startCase from 'lodash/startCase'
+import _flatMap from 'lodash/flatMap'
+import Papa from "papaparse";
+
+/*
+Using the DataTable
+Required Props: columns, data
+Optional Props: allowExport (default: true)
+
+Columns:
+Array of objects consisting of a field and header at minimum.
+- Field must match the key of the entry in the data object.
+- Header is an optional string that is displayed at the top of 
+      the column.
+- dataType is a string that defines the data type of the column.
+      options are TEXT, NUMERIC, or DATE
+- Sort (optional) is a boolean field that determines whether sorting
+      is to be allowed on the field. If it is not present, defaults to true.
+- allowMultipleFilters (optional) is a boolean field that determines whether
+      users have the option of apply multiple filters.
+- useMultiSelect is an optional boolean field that determines whether the
+      filter will be a multi-select dropdown. options are generated by the 
+      given data.
+*/
+
+const props = defineProps({
+  columns: {type:Array, required: true},
+  data: {type: Array, required: true},
+  allowExport: {type: Boolean, default: true}
+});
+
+let selectedRows = [];
+const refSelectedRows = ref(selectedRows);
+
+const exportCSV = async () => {
+  const csv = Papa.unparse(_flatMap(refSelectedRows.value, prepareData));
```

Comment:
```suggestion
  const csv = Papa.unparse(_flatMap(refSelectedRows.value, prepareDates));
```

See comment below about function name.